fvm/src/call_manager/default.rs

```diff
@@ -432,11 +432,33 @@ where
 
             // From this point on, there are no more syscall errors, only aborts.
             let result: std::result::Result<BlockId, Abort> = (|| {
+                use wasmtime_runtime::InstantiationError;
                 // Instantiate the module.
                 let instance = engine
                     .get_instance(&mut store, &state.code)
                     .and_then(|i| i.context("actor code not found"))
-                    .map_err(Abort::Fatal)?;
+                    .map_err(|e| match e.downcast::<InstantiationError>() {
+                        Ok(e) => match e {
+                            // This will be handled in validation.
+                            InstantiationError::Link(e) => Abort::Fatal(anyhow!(e)),
+                            // TODO: We may want a separate OOM exit code? However, normal ooms will usually exit with SYS_ILLEGAL_INSTRUCTION.
```

Comment:
precedent in UNIX says we probably dont.
Would it help anyone?

We should definitely log this loudly however.

Reply:
>  Would it help anyone?

It may help with some debugging. Let's just leave these TODOs here and move on for now.